### PR TITLE
Fix/xcode build

### PR DESCRIPTION
**Pull request by BlueCocoa**
IMHO, using cmake to generate Xcode project would be better. Imagine if someone uses Linux and writes some code for this project, then he has to edit Xcode project file by hand. 
Even if he leaves us to do this job, it's a little bit complex to maintain Xcode project by hand.

Perhaps a vote?
(+1 for using cmake, -1 for maintaining by hand)

@al1enSuu @Sunbelife @Giolno @ENzero @hem1

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/Llama-Group/Project-Llama/pull/79" target="_blank">ReviewNinja

**Comment by 0xsuu** [comment]
0

File: README.md [hunk 1]
```diff
@@ -16,3 +16,9 @@ Installation
 ```bash
 git clone --recursive https://github.com/Llama-Group/Project-Llama.git
 ```
+
+### Xcode build
+```
+cd llama_cpp/
+cmake -G Xcode
```
Comment (0xsuu) [inline README.md]: Why build here?

File: build/build_cpp.sh [hunk 2]
```diff
@@ -7,15 +7,28 @@ BASEDIR=$(dirname "$0")
...
+	mkdir -p ../build_llama_cpp
+	cd ../build_llama_cpp
+	
 	# Build library
 	cmake ../Project-Llama/llama_cpp
+elif [ "$1" == "Xcode" ]; then
+	# Create Xcode project folder
+	rm -rf ../llama_xcode
+	mkdir -p ../llama_xcode
+	cd ../llama_xcode
```
Comment (0xsuu) [inline build/build_cpp.sh]: Use pushd and popd so below code won't be affected.

File: build/build_cpp.sh [hunk 3]
```diff
@@ -63,12 +76,24 @@ else
...
+if [ "$1" == "Xcode" ]; then
+	case `uname` in 
+		Darwin*) xcodebuild && open ./;;
+		**) echo "Xcode Project generated at ../llama_xcode/llama_cpp.xcodeproj" ;;
+	esac
+else
+	make
+fi
+
 if [ $? != 0 ]; then
```
Comment (0xsuu) [inline build/build_cpp.sh]: Move this "if" into above else case. And write conditions for Xcode generation failed.

File: build/build_cpp.sh [hunk 4]
```diff
@@ -63,12 +76,24 @@ else
 	else
 		echo "Usage: ./build_cpp.sh 	[-e|--example] [example name]|all"
 		echo "			[-b|--benchmark] [benchmark name]|all"
+		case `uname` in
```
Comment (0xsuu) [inline build/build_cpp.sh]: Consider using uname -s

File: build/build_cpp.sh (hunk 2 above)
Comment (0xsuu) [inline build/build_cpp.sh]: Can also save some lines of codes as Xcode is a side building target, except from that, every building things should be in build_llama_cpp.

**Comment by 0xsuu** [comment]
LGTM please add instructuon in wiki-quick start

**Comment by 0xsuu** [comment]
LGTM

**Comment by 0xsuu** [comment]
LGTM

**Pull request merged**